components/DragAndDropInput.vue

```diff
@@ -0,0 +1,267 @@
+
+const maxFileSizeBytes = computed(() => {
+  return props.maxFileSizeMB * 1000000;
+});
+
+const resetState = () => {
+  message.value = "";
+  uploadedFiles.value = [];
+  notUploadedFiles.value = [];
+};
+
+const handleDrag = (e: DragEvent) => {
+  if (e.type === "dragenter" || e.type === "dragover") {
+    isDragActive.value = true;
+    resetState();
+  }
+  if (e.type === "dragleave") {
+    isDragActive.value = false;
+  }
+};
+
+const numberOfFilesExceeded = (
+  listOfFiles: FileList | DataTransferItemList
+) => {
+  if (listOfFiles.length > props.maxFilesNumber) {
+    message.value = `You can upload up to ${props.maxFilesNumber} files at a time!`;
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const handleDrop = (e: DragEvent) => {
+  e.stopPropagation();
+  isDragActive.value = false;
+
+  if (e.dataTransfer?.items) {
+    if (numberOfFilesExceeded(e.dataTransfer.items)) return;
+    [...e.dataTransfer.items].forEach((item) => {
+      if (item.kind === "file") {
+        const file = item.getAsFile();
+        if (file) {
+          if (file.size < maxFileSizeBytes.value) {
+            uploadedFiles.value.push(file.name);
+            // send to backend
+          } else {
+            notUploadedFiles.value.push(file.name);
+          }
+        }
+      }
+    });
+  } else if (e.dataTransfer?.files) {
+    if (numberOfFilesExceeded(e.dataTransfer.files)) return;
+    [...e.dataTransfer.files].forEach((file) => {
+      if (file.size < maxFileSizeBytes.value) {
+        uploadedFiles.value.push(file.name);
+        // send to backend
+      } else {
+        notUploadedFiles.value.push(file.name);
+      }
+    });
+  }
+};
+
+const handleUpload = (e: Event) => {
+  const target = e.target as HTMLInputElement;
+  if (!target.files || numberOfFilesExceeded(target.files)) return;
+  [...target.files].forEach((file) => {
+    if (file.size < maxFileSizeBytes.value) {
+      uploadedFiles.value.push(file.name);
+      // send to backend
+    } else {
+      notUploadedFiles.value.push(file.name);
+    }
+  });
+};
+
+const handleKeydown = (e: KeyboardEvent) => {
+  const input = root.value?.querySelector("#file") as HTMLInputElement | null;
+  if (e.code === "Enter") {
+    resetState();
+    input?.click();
+  }
+};
+</script>
+
+<template>
+  <div ref="root" class="wrapper">
+    <form class="file--form" @submit.prevent @dragenter="handleDrag">
+      <p class="file--info">
+        You can upload max {{ maxFilesNumber }} files, max {{ maxFileSizeMB }}MB
+        each.
+      </p>
+      <div
+        class="file--dropzone"
+        :class="{ active: isDragActive }"
+        @drop.prevent="handleDrop"
+        @dragenter="handleDrag"
+        @dragover.prevent="handleDrag"
+        @dragleave="handleDrag"
+      >
+        <p class="file--dropzone-description">Drop your files here</p>
+        <img
+          class="file--dropzone-icon"
+          src="../assets/img/drag-and-drop.png"
```

Comment:
it is safer to use absolute path to assets if possible - it's implemented out of the box in Nuxt and you can set `@/` instead of `../` here 🧑‍🍳 

It's safer cause you don't need to remember of this import if you will need to move component in some other place 👌